### PR TITLE
Implement classes for CAS-CVAT integration

### PR DESCRIPTION
**Pull request by ptelang**
This commit adds classes in support of CAS-CVAT integration.

### CODE REVIEW

**Comment by ptelang** [comment]
@courtneyambrozic, @fijoy - Can you please review this PR? Thanks.

**Review by courtneyambrozic — APPROVED** [review]
Hi Pankaj, awesome job with this. I'll defer to @fijoy for his approval before we get this merged.

File: cvpy/annotation/base/Project.py
Comment (fijoy) [inline cvpy/annotation/base/Project.py:18]: What's the difference between project_name and project_id? Can we have just the name as a parameter, and generate an ID inside the class for internal use that the user does not need to see?

File: cvpy/annotation/base/Project.py
Comment (fijoy) [inline cvpy/annotation/base/Project.py:79]: I'm not entirely familiar with the decorators here, but it looks like you have already implemented setters and getters. Shruti had a story to design these implementations. Can you check with her to see if anything is missing? Make sure you're on the same page, and we can delete any future stories that Shruti has created related to this.

File: cvpy/annotation/base/Credentials.py
Comment (fijoy) [inline cvpy/annotation/base/Credentials.py:8]: Should it be "user" or "username"? Or something else like "login"? I haven't seen just "user" used in authentication.

File: cvpy/annotation/cvat/CVATProject.py
Comment (fijoy) [inline cvpy/annotation/cvat/CVATProject.py:97]: Thanks for laying down the skeleton for these methods.

File: cvpy/tests/test_credentials.py
Comment (fijoy) [inline cvpy/tests/test_credentials.py:58]: Can we have a brief comment right before the definitions of each of these tests describing what it's actually testing? This will be similar to how we comment unit tests in C: https://grok.sas.com/xref/mva-vbviya/tkcv/test/tkimcore/src/imageSobel.c#15

File: cvpy/tests/test_credentials.py
Comment (fijoy) [inline cvpy/tests/test_credentials.py:73]: This is the first time I'm reviewing CVPy change. Will these tests be run automatically in future pushes? Or do we need to  do something else to make that happen, like modifying a Jenkins file? Not that we're about to add a lot to CVPy, we should make sure existing tests are being run as code is updated.

File: cvpy/tests/test_cvat_project.py
Comment (fijoy) [inline cvpy/tests/test_cvat_project.py:63]: Moving forward, will these tests be run automatically against our team-wide CVAT server cvdata.unx.sas.com as we push new features?

File: cvpy/tests/test_cvat_project.py
Comment (fijoy) [inline cvpy/tests/test_cvat_project.py:48]: Same commend as before. Can we insert a brief comment above each of these test functions?

**Review by fijoy — COMMENTED** [review]
Thanks Pankaj for working on this. The layout and content look great overall. 

I have several mostly minor comments for your to address.

File: cvpy/annotation/base/Credentials.py
Comment (ptelang) [reply cvpy/annotation/base/Credentials.py]: Done

Comment (ptelang) [reply cvpy/annotation/base/Credentials.py]: Done

File: cvpy/annotation/base/Project.py
Comment (ptelang) [reply cvpy/annotation/base/Project.py]: Good point. I removed project_id from the constructor parameters.

File: cvpy/annotation/base/Project.py
Comment (ptelang) [reply cvpy/annotation/base/Project.py]: I am checking with Shruti.

File: cvpy/annotation/base/Credentials.py
Comment (ptelang) [reply cvpy/annotation/base/Credentials.py]: I changed it to username which is consistent with swat CAS connection parameter.

File: cvpy/annotation/cvat/CVATProject.py
Comment (ptelang) [reply cvpy/annotation/cvat/CVATProject.py]: Ack

File: cvpy/tests/test_credentials.py
Comment (ptelang) [reply cvpy/tests/test_credentials.py]: Done

File: cvpy/tests/test_cvat_project.py
Comment (ptelang) [reply cvpy/tests/test_cvat_project.py]: Done

File: cvpy/tests/test_credentials.py
Comment (ptelang) [reply cvpy/tests/test_credentials.py]: I will check with @courtneyambrozic.

Comment (courtneyambrozic) [reply cvpy/tests/test_credentials.py]: Yes, we'll have to modify the Jenkinsfile to run these new tests: https://gitlab.sas.com/computer-vision/devops/jenkins/cvpy-pipeline/-/blob/master/Jenkinsfile

Mu is adding a new test file too, so maybe we can create a story to modify the Jenkinsfile to run both of these.

The Jenkins pipeline isn't set up to run for every new pull request (this is something that we discussed early on but posed a potential security risk). Currently, we just run the Jenkins job manually using the PR number as an input and then again when the PR is merged.

File: cvpy/annotation/cvat/CVATProject.py
Comment (courtneyambrozic) [inline cvpy/annotation/cvat/CVATProject.py:101]: Maybe we should first push this image table class (and other relevant files) to main so that it can be used by Mu and Kevin. Do you have any thoughts on this, Pankaj?

Comment (ptelang) [reply cvpy/annotation/cvat/CVATProject.py:101]: That is a great idea. I just created a separate PR just for pushing ImageTable class. Can you please review it? Thanks.

File: cvpy/tests/test_cvat_project.py
Comment (ptelang) [reply cvpy/tests/test_cvat_project.py]: The CVAT url is an argument to the test, which we will set to the team-wide CVAT server (for now). So all unit tests even for new features will run against this server.

File: cvpy/tests/test_credentials.py
Comment (ptelang) [reply cvpy/tests/test_credentials.py]: We should not need to modify Jenkins file as we add more classes and unit tests in the future. I created a ticket (COMPVIS-4123) to address this issue.

File: cvpy/annotation/base/Project.py
Comment (ptelang) [reply cvpy/annotation/base/Project.py]: Shruti pointed out a missing function for getting the tasks. I added that function and a supporting Task class. So Shruti can now delete the other story. I will let her know.

**Pull request merged**
